src/shogun/classifier/mkl/MKLOneClass.cpp

```diff
@@ -20,6 +20,6 @@ float64_t CMKLOneClass::compute_sum_alpha()
 
 void CMKLOneClass::init_training()
 {
+	ASSERT(dynamic_cast<CLibSVMOneClass*>(svm) != nullptr)
```

Comment:
dynamic casting looks ugly to me. previous line is much cleaner
`ASSERT(svm->get_classifier_type() == CT_LIBSVMONECLASS)`

Reply:
The issue is that we want to go for a plugin architecture of shogun soon. This means that the specialization of classes are not available/visible to the base classes anymore. Only base class interfaces will be. Therefore, this stuff will need to be communicated via the API or `get` calls

Reply:
This sounds much cleaner! So this should allow people to create their custom SVMs for example using some factory interface?

Reply:
Exactly, new SVMs would just have to serve the base class API. This is the reason why we cannot have central structures for those plugins/classes anymore -- they will only be known at runtime

Reply:
I use the name instead of the enum now
```C++
ASSERT(std::string(svm->get_name()) == "LibSVMOneClass")
```

Reply:
ah I am not so sure about this. I know it works but it is asking for trouble. As far as I can seem the SVM is never explicitly casted to LibSVMOneClass (or am I wrong?) So I would really prefer some method in `CSVM` that indicates support for whatever the MKL class wants to do/check.

Reply:
I can add a method that indicates one_class, but then `SVMLightOneClass` should also satisfy this, and it's not supported judging from this line 'ASSERT(svm->get_classifier_type() == CT_LIBSVMONECLASS)'

Reply:
The reasons why this assertion is in here seem dubious to me. What you say makes sense, checking that the SVM is one class. So let's just add that method to CSVM.